pkg/recorder/diskrecorder/diskrecorder.go

```diff
@@ -44,6 +44,29 @@ func (d *DiskRecorder) SaveAtPath(records record.MemoryRecords, path string) (re
 		return nil, fmt.Errorf(`path should have suffix "%v"`, archiveExtension)
 	}
 
+	f, err := os.OpenFile(path, os.O_CREATE|os.O_WRONLY|os.O_EXCL, 0640)
+	if err != nil {
+		if os.IsExist(err) {
+			klog.Errorf("Tried to copy to %s which already exists", path)
+		}
+		return nil, fmt.Errorf("nable to create archive: %v", err)
+	}
+
+	klog.Infof("Writing %d records to %s", len(records), path)
+
+	completed, err := writeRecordsToArchive(records, f)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := f.Close(); err != nil {
```

Comment:
Why isn't this called with `defer`? If it returns on the line 52 (or 59) then the file is never closed.  I would recommend to check the test failures and also extend the unit tests.